### ISSUE TITLE
[BUG] Abstract components are missing from output

### ISSUE THREAD

**Issue by alastair-todd** [compodoc/compodoc#957]
Abstract components are not included by compodoc. Is it because they are not added to a Module perhaps? After Ivy, ng 9+, we have to decorate anything that is component-like with a Directive attribute.

Anyway it can't be added to a module. Is there a way around this?

````typescript
/**
* abstract components are not included
*/
@Directive()
// tslint:disable-next-line:directive-class-suffix
export abstract class AbstractComponent implements OnInit {
  ngOnInit() {}
}
````

**Comment by BeSpunky** [comment]
Hi guys, any news on this?

The abstract classes I provide in my library are core features and they are not documented... Very frustrating.

**Comment by BeSpunky** [comment]
Found a possible workaround:

Decorate your abstract class as BOTH `@Injectable()` and `@Directive()`.
Angular doesn't complain, my tests keep passing and the class appears in the `Injectables` section.

❕ Not sure if there are any Angular side effects for a class decorated with both.
In a small POC stackblitz I made everything worked as expected, no errors, nothing breaks.
Please comment if you know of any side effect.

Cheers 🥂

**Comment by vogloblinsky** [comment]
@jgomesmv do you have an example project ?

**Comment by BeSpunky** [comment]
### Repo
https://github.com/BeSpunky/angular-zen

### Decorated Class
https://github.com/BeSpunky/angular-zen/blob/e320064f964a029c5f3b80da7ccafb230dd3e9df/projects/bespunky/angular-zen/router-x/services/route-aware.service.ts#L53-L54

### Generated Doc
https://bs-angular-zen.web.app/docs/zen/injectables/RouteAware.html

### Compodoc Config
https://github.com/BeSpunky/angular-zen/blob/530225fc9fe19f607300b778a06f39a341009992/projects/bespunky/angular-zen/.compodocrc

### NPM Scripts
https://github.com/BeSpunky/angular-zen/blob/530225fc9fe19f607300b778a06f39a341009992/package.json#L24-L25

**Comment by literalpie** [comment]
Will this be fixed?

**Comment by vogloblinsky** [comment]
> Will this be fixed?

I look at @BeSpunky example right now

**Comment by vogloblinsky** [comment]
Ok, @BeSpunky, fixed with next release, 1.1.12. Remove the @directive for both failing files LocalizedRouteAware and RouteAware
